packages/dialog/README.md

```diff
@@ -0,0 +1,397 @@
+| `--mdc-dialog-max-width`            | `560px`               | max-width of the dialog surface. (**Note:** if max-width is < `560px`, there is a visual jank bug that will occur causing the max width to be `560px` when the window is sized to <= than `560px`).
+| `--mdc-dialog-max-height`           | `calc(100% - 32px)`   | Max height of the dialog surface.
+| `--mdc-dialog-shape-radius`         | `4px`                 | Corner radius of the dialog surface.
+
+### Actions
+
+Actions close the dialog on click. You can define an action by slotting an
+element with the `dialog-action="..."` string attribute. The name of the
+attribute can be customized by the
+[`actionAttribute` property](#Properties/Attributes). When a clickable element
+with the `dialog-action` attribute is clicked, `mwc-dialog` will get the value
+of the attribute and fire the `closing` and subsequent `closed` events with a
+detail of `{action: <clickedElement.getAttribute('dialog-action')>}`.
+
+For example:
+
+<img src="images/action.png" width="566px">
+
+```html
+<mwc-dialog open>
+  <div>
+    <div>
+      This is my content. Here is an actionable button:
+      <button dialog-action="contentButton">button 1</button>
+    </div>
+    <div>
+      This is my content. Here is a diabled actionable button:
+      <button disabled dialog-action="disabledContentButton">button 2</button>
+    </div>
+  </div>
+  <mwc-button slot="primaryAction" dialog-action="ok">ok</mwc-button>
+  <mwc-button slot="secondaryAction">cancel</mwc-button>
+</mwc-dialog>
+```
+
+In this example we have 3 actionable elements:
+```html
+<button dialog-action="contentButton">button 1</button>
+```
+
+```html
+<button disabled dialog-action="disabledContentButton">button 2</button>
+```
+
+```html
+<mwc-button slot="primaryAction" dialog-action="ok">ok</mwc-button>
+```
+
+* Clicking button 1 will close the dialog and fire a `closing` and subsequently
+a `closed` event with a detail of `{action: 'contentButton'}`.
+* Clicking button 2 will not close the dialog since it is disabled
+* Clicking the cancel `mwc-button` will not close the dialog as it does not have
+a `dialog-action` attribute set on it.
+* Clicking the ok `mwc-button` will close the dialog and fire a `closing` and
+subsequently a `closed` event with a detail of `{action: 'ok'}`.
+* Setting `document.querySelector('mwc-dialog').open = false;` will close the
+dialog and fire a `closing` and subsequently a `closed` event with a detail of
+`{action: 'close'}` (action is configurable via
+[`defaultAction` property](#Properties/Attributes)).
+
+### Focus
+
+Initial focus can be set on an element with the `dialog-initial-focus` boolean
+attribute (configurable via the
+[`initialFocusAttribute` property](#Properties/Attributes)).
+
+For example:
+
+<img src="images/initial-focus.png" width="594x">
+
+```html
+<mwc-dialog title="Initial Focus">
+  <div>
+    In this example we set "dialog-initial-focus" on a focusable element.
```

Comment:
Kinda a nit, but the "in this example we" explanation is rendered 3 times (in the screenshot, in the code, and in the markdown below). I think the example could just have a generic message, because it's slightly jarring to see it 3 times (plus it's a little different the 3rd time).

Reply:
I am 100% on board with hammering it home!